internal/chain/engine.go

```diff
@@ -324,6 +329,44 @@ func (e *engineImpl) Finalize(
 			}
 		}
 	}
+
+	if inStakingEra {
+		if isBeaconChain {
```

Comment:
should also add isNewEpoch, because we should only do the tallying at the end of epoch. And with this, we should put the tallying before the shard state proposal, so kicked out nodes won't be part of shard anymore.